edsnlp/pipelines/misc/measurements/measurements.py

```diff
@@ -0,0 +1,669 @@
+            List of tuple of span and whether the span represents a sentence end
+        pseudo_mapping: Dict[int, str]
+            A mapping from label to char in the pseudo sentence
+
+        Returns
+        -------
+        (str, List[int])
+            - the pseudo sentence
+            - a list of offsets to convert match indices into pseudo sent char indices
+        """
+        pseudo = []
+        last = 0
+        offsets = []
+        for ent, is_sent_split in matches:
+            if ent.start != last:
+                pseudo.append("w")
+            offsets.append(len(pseudo))
+            if is_sent_split:
+                pseudo.append(".")
+            else:
+                pseudo.append(pseudo_mapping.get(ent.label, "w"))
+            last = ent.end
+        if len(doc) != last:
+            pseudo.append("w")
+        pseudo = "".join(pseudo)
+
+        return pseudo, offsets
+
+    def get_matches(self, doc):
+        """
+        Extract and filter regex and phrase matches in the document
+        to prepare the measurement extraction.
+        Returns the matches and a list of hashes to quickly find unit matches
+
+        Parameters
+        ----------
+        doc: Doc
+
+        Returns
+        -------
+        Tuple[List[(Span, bool)], Set[int]]
+            - List of tuples of spans and whether the spans represents a sentence end
+            - List of hash label to distinguish unit from other matches
+        """
+        sent_ends = [doc[i : i + 1] for i in range(len(doc)) if doc[i].is_sent_end]
+
+        regex_matches = list(self.regex_matcher(doc, as_spans=True))
+        term_matches = list(self.term_matcher(doc, as_spans=True))
+
+        # Detect unit parts and compose them into units
+        units = self.extract_units(term_matches)
+        unit_label_hashes = {unit.label for unit in units}
+
+        # Filter matches to prevent matches over dates or doc entities
+        non_unit_terms = [
+            term
+            for term in term_matches
+            if term.label not in self.unit_part_label_hashes
+        ]
+
+        # Filter out measurement-related spans that overlap already matched
+        # entities (in doc.ents or doc.spans["dates"])
+        # Note: we also include sentence ends tokens as 1-token spans in those matches
+        spans__keep__is_sent_end = filter_spans(
+            [
+                # Tuples (span, keep = is measurement related, is sentence end)
+                *zip(doc.spans.get("dates", ()), repeat(False), repeat(False)),
+                *zip(regex_matches, repeat(True), repeat(False)),
+                *zip(non_unit_terms, repeat(True), repeat(False)),
+                *zip(units, repeat(True), repeat(False)),
+                *zip(doc.ents, repeat(False), repeat(False)),
+                *zip(sent_ends, repeat(True), repeat(True)),
+            ],
+            # filter entities to keep only the ...
+            sort_key=lambda span__keep__is_sent_end: (
```

Comment:
I was looking for a more explicit way of doing this, and stumbled upon this little nugget: https://peps.python.org/pep-3113/

legibility would benefit from using a function definition on this one. To wit, something like:

```python
def sort_tuples(span__keep__is_sent_end: Tuple[Span, bool, bool]) -> Tuple[int, int, bool]:
    span, _, is_sent_end = span__keep__is_sent_end

    length = span.end - span.start

    return (length, span.end, not is_sent_end)
```